vividus-plugin-rest-api/src/main/java/org/vividus/bdd/steps/api/JsonResponseValidationSteps.java

```diff
@@ -233,16 +241,23 @@ public void waitForJsonFieldAppearance(String jsonPath, String resourceUrl, Dura
         long pollingTimeoutMillis = durationInMillis / retryTimes;
         httpRequestExecutor.executeHttpRequest(HttpMethod.GET, resourceUrl, Optional.empty(), response ->
         {
-            if (response == null || getElementsNumber(response.getResponseBodyAsString(), jsonPath) > 0)
+            try
+            {
+                if (response == null || getElementsNumber(response.getResponseBodyAsString(), jsonPath) > 0)
+                {
+                    return false;
+                }
+            }
+            catch (InvalidJsonException ignored)
             {
-                return false;
+                return true;
```

Comment:
what will happen if the endpoint will always return response with non JSON content type? it looks like it will become an infinite loop, won't it?

Reply:
also there is no `sleep` after response with non JSON content type

Reply:
No need to return something there, I will exclude the return statement.